service/debugger/debugger.go

```diff
@@ -1449,6 +1542,70 @@ func traverse(t proc.ValidTargets, f *proc.Function, depth int, followCalls int)
 	return funcs, nil
 }
 
+// createBreakpointForTracepoints creates a logical breakpoint specifically for tracepoints.
+// It handles the BreakpointExistsError by silently ignoring it, as duplicate tracepoints
+// are expected during dynamic function discovery.
+func (d *Debugger) createBreakpointForTracepoints(lbp *proc.LogicalBreakpoint) error {
+	d.breakpointIDCounter++
+	lbp.LogicalID = d.breakpointIDCounter
+	lbp.HitCount = make(map[int64]uint64)
+	d.target.LogicalBreakpoints[lbp.LogicalID] = lbp
+	err := d.target.SetBreakpointEnabled(lbp, true)
+	if err != nil {
+		delete(d.target.LogicalBreakpoints, lbp.LogicalID)
+		// Silently ignore BreakpointExistsError - this is expected when
+		// creating tracepoints for functions that may already have them
+		if _, exists := err.(proc.BreakpointExistsError); exists {
+			return nil
+		}
+		return err
+	}
+	return nil
+}
+
+// createFnTracepoint is a way to create a trace point late in the cycle but just in time that it can be
+// included in the trace output as in the case of dynamic functions, we get to know the functions that are
+// being called from deferreturn quite late in execution
+func createFnTracepoint(d *Debugger, fname string, rootstr string, followCalls int) (*api.Breakpoint, error) {
```

Comment:
Sorry to be pedantic, but technically this creates multiple breakpoints (entry, exit), so maybe we rename this to `createFunctionTracepoints` (note plural).